### PR TITLE
feat: implement home

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

File: examples/home/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,29 @@
+load("//distroless:defs.bzl", "home")
+load("//distroless/tests:asserts.bzl", "assert_tar_listing")
+
+home(
+    name = "home",
+    dirs = [
+        {
+            "home": "./root",
```
Comment (loosebazooka) [inline examples/home/BUILD.bazel:8]: using `./` feels odd to me, it should be at root? And this make it seems like it's not?

File: distroless/private/tar.bzl [hunk 2]
```diff
@@ -15,21 +15,29 @@ def _mtree_line(file, type, content = None, uid = "0", gid = "0", time = "167256
         spec.append("content=" + content)
     return " ".join(spec)
 
-def _add_parents(path):
+def _add_parents(path, uid = "0", gid = "0", time = "1672560000", mode = "0755"):
```
[the remark this answers is not in the document]
Comment (thesayyn) [reply distroless/private/tar.bzl:18]: Right, it's copy-pasta from bazel-lib. will make it 0.

File: distroless/private/home.bzl [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+load("@aspect_bazel_lib//lib:tar.bzl", "tar")
+load(":tar.bzl", "tar_lib")
+
+def home(name, dirs, **kwargs):
+    """
+    Create home directories with specific uid and gids.
+
+    Args:
+        name: name of the target
+        dirs: an array of dicts
```
[the remark this answers is not in the document]
Comment (thesayyn) [reply distroless/private/home.bzl:12]: agreed

**Review by loosebazooka — COMMENTED** [review]
I think this PR still has the 167256000 timestamp, change here or in a followup?

Also will home be auto integrated into passwd entry files? for users? Or is this something that needs to be created in addition to defining users and groups?

**Review by loosebazooka — APPROVED** [review]

**Comment by thesayyn** [comment]
> I think this PR still has the 167256000 timestamp, change here or in a followup?
> 
> Also will home be auto integrated into passwd entry files? for users? Or is this something that needs to be created in addition to defining users and groups?

That’ll change in a follow up. Home will be used with passwd and flatten in a single tar for performance.

**Pull request merged**